hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/WrappedHederaState.java

```diff
@@ -71,16 +71,16 @@ public ReadableStates createReadableStates(@NonNull String serviceName) {
      */
     @Override
     @NonNull
-    public WritableStates createWritableStates(@NonNull String serviceName) {
+    public HederaWritableStates createWritableStates(@NonNull String serviceName) {
         return writableStatesMap.computeIfAbsent(
-                serviceName, s -> new WrappedWritableStates(delegate.createWritableStates(s)));
+                serviceName, s -> (HederaWritableStates) delegate.createWritableStates(s));
```

Comment:
This looks wrong. Do we really need both interfaces, `HederaWritableStates` and `WritableStates`? What do you think about merging them?

Reply:
I have reverted this and have a better solution suggested by @tinker-michaelj. This change I have here would actually be bad, in that a commit on some arbitrary point in the save stack would go straight to the merkle tree! Fortunately, the unit tests failed loudly, so it was caught.